### ISSUE TITLE
[K2+SE] Crash while attempting to edit beacon module

### ISSUE THREAD

**Issue by the-daveman** [ClaudeMetz/FactoryPlanner#261]
### Problem Description

I simply attempted to switch the beacon module between different levels of speed modules to see the difference in machine count I would need for my blueprint. I changed it from Speed6 to Speed5. Then when attempting to change it back to Speed6, the game instantly crashes when I attempt to edit it.
Screenshots attached.
Modlist attached.

### Reproduction

I simply load the game, open factory planner.
Select the beacon module for the Refinery, right click, and the game instantly crashes with the error message.

```
Error while running event factoryplanner::on_gui_click (ID 1)
__factoryplanner__/ui/main/production_handler.lua:168: attempt to index local 'module' (a nil value)
```
![427520_20240213211719_1](https://github.com/ClaudeMetz/FactoryPlanner/assets/5562517/a9e9dee9-8465-4bb1-97f5-7ab33f20da88)
![427520_20240213211726_1](https://github.com/ClaudeMetz/FactoryPlanner/assets/5562517/4edeec48-2570-4380-ad1d-46b17c9ee022)

I can upload and link to the savegame zip if requested.

**Comment by the-daveman** [comment]
Seems like the same issue with similar reproduction steps to this bug:

https://github.com/ClaudeMetz/FactoryPlanner/issues/263

**Comment by ClaudeMetz** [comment]
Appreciate the report. This was incidentally already fixed for 2.0, which is very close now :)
